crates/spirv-std/src/math_ext.rs

```diff
@@ -82,4 +100,24 @@ impl MathExt for f32 {
     fn copysign(self, sign: f32) -> f32 {
         unsafe { core::intrinsics::copysignf32(self, sign) }
     }
+
+    fn step(self, edge: f32) -> f32 {
+        if self < edge {
+            0.0
+        } else {
+            1.0
+        }
+    }
+
+    fn step_select(self, edge: f32, less: f32, greater_or_equal: f32) -> f32 {
+        if self < edge {
+            less
+        } else {
+            greater_or_equal
+        }
+    }
+
+    fn mix(self, other: f32, a: f32) -> f32 {
```

Comment:
I think I'd prefer this to be `lerp`, but that's mostly bikeshedding and I don't really care. Also, I think I'd prefer `t.lerp(a, b)` instead of `a.lerp(b, t)`, although Freya's a lil sketchy here and thinks that both ways are valid interpretations, and almost wouldn't want an "object"-style method to avoid that ambiguity, haha. (`t.lerp(a, b)` makes sense in shaders, `a.lerp(b, t)` makes sense when animating objects)

Reply:
Yeah, I originally agreed with that, but one thing that one of your comments brought up actually is that I feel like `lerp` has a bit more of a stringent mathematical definition/interpretation that doesn't really allow something like `fn lerp(self: Vec3, other: Vec3, a: Vec3)` whereas with a more loosey-goosey definition of `mix` it can make sense (and is sometimes useful).

Reply:
I'm a little torn on this one, but I think if we consider inverse lerp, I think it's crystal clear that that the value parameter should go first, because in 99% of cases, you are remapping a value into a 0-1 range

`t = InvLerp( a, b, value )`
`t = value.InvLerp( a, b )`

and looking back at lerp, we're usually blending between two states, and having those states next to each other to signify the "input range" makes a lot of sense

`v = Lerp( a, b, t )`
`v = t.Lerp( a, b )`

And with the "remapping to a range" perspective, this one too is remapping a t value into another range. The one thing that makes me uncomfortable with this one though is that t is usually a float, but it can return a vector (or vice versa) which feels a little spooky to me for some reason~ But in the context of value remapping it kinda makes sense. Just like you could have a remap function which would read pretty naturally too:

`vOut = vIn.Remap( iMin, iMax, oMin, oMax )`

Reply:
or be really spicy with `t.mix(a..b)` >:D (/s. mostly.)